### PR TITLE
[Docs] Add license badge and Twitter follow button to top-level README.md

### PR DESCRIPTION
**Pull request by feluelle**
Description:

I think it would be nice when the twitter page is mentioned. :)
I also added a license badge to link to the official location of the apache license.

Changes:

- add license badge
- add twitter badge
- change build and coverage badges to use markdown instead of html
- change headlines to be one level bigger

### CODE REVIEW

**Comment by feluelle** [comment]
I would also add the docs, but I am not sure how to link the badge correctly so it links to this https://arrow.apache.org/docs/ and shows the badge.

**Comment by wesm** [comment]
I'm curious about the "ARROW-XXX". It's not necessary to have a JIRA for an edit to the README, but either remove that from the PR title or open a JIRA issue and add the issue number to the PR title

File: README.md [hunk 1]
```diff
@@ -17,26 +17,14 @@
...
-    </td>
-  </tr>
-</table>
-
-### Powering In-Memory Analytics
+# Apache Arrow
+
+[![Build Status](https://travis-ci.org/apache/arrow.svg?branch=master)](https://travis-ci.org/apache/arrow)
+[![Coverage Status](https://codecov.io/gh/apache/arrow/branch/master/graph/badge.svg)](https://codecov.io/gh/apache/arrow?branch=master)
+[![License](http://img.shields.io/:license-Apache%202-blue.svg)](http://www.apache.org/licenses/LICENSE-2.0.txt)
```
Comment (wesm) [inline README.md]: We should link to our license file, which attributes third party licenses that are utilized in the codebase

**Review by wesm — APPROVED** [review]
+1

File: README.md (hunk 1 above)
Comment (feluelle) [reply README.md]: Makes sense. 👍

**Comment by feluelle** [comment]
> I'm curious about the "ARROW-XXX". It's not necessary to have a JIRA for an edit to the README, but either remove that from the PR title or open a JIRA issue and add the issue number to the PR title

Sorry, this is my second project I contribute to 😊 . Did not read the contributing guide probably. In Apache Airflow it was AIRFLOW-XXX.. so I guess this was the reason.

Anyway thank you for accepting my PR :)

@wesm What do you actually think about adding a docs badge, too?

**Comment by wesm** [comment]
I think maybe you read the "XXX" a bit too literally -- that is supposed to be filled in with the actual JIRA number from https://issues.apache.org/jira. See https://github.com/apache/airflow/blob/master/CONTRIBUTING.md#pull-request-guidelines

> "Preface your commit's subject & PR's title with [AIRFLOW-XXX] where XXX is the JIRA number."

**Comment by wesm** [comment]
If you want to add a docs badge please go ahead

**Comment by feluelle** [comment]
> In case you are fixing a typo in the documentation you can prepend your commit with [AIRFLOW-XXX], code changes always need a Jira issue.

Do I also understand this wrong? (from the PR template) And everyone is doing it by prepending AIRFLOW-XXX if it is a change to the docs.
I just want to understand if it is really only a template or if it is maybe for the jira Integration to just differ doc PRs from code PRs.

**Comment by wesm** [comment]
I see. I guess that is an Airflow thing. I don't think we intend to adopt that convention, it looks ugly and unuseful to me

**Comment by feluelle** [comment]
+1, I actually like [docs] more.

**Pull request merged**